BrowserKit/Sources/ComponentLibrary/Buttons/SecondaryRoundedButton.swift

```diff
@@ -5,54 +5,85 @@
 import Common
 import UIKit
 
-public class SecondaryRoundedButton: LegacyResizableButton, ThemeApplicable {
+public class SecondaryRoundedButton: ResizableButton, ThemeApplicable {
     private struct UX {
         static let buttonCornerRadius: CGFloat = 12
         static let buttonVerticalInset: CGFloat = 12
         static let buttonHorizontalInset: CGFloat = 16
         static let buttonFontSize: CGFloat = 16
-    }
-
-    private var highlightedTintColor: UIColor!
-    private var normalTintColor: UIColor!
 
-    override open var isHighlighted: Bool {
-        didSet {
-            backgroundColor = isHighlighted ? highlightedTintColor : normalTintColor
-        }
+        static let contentInsets = NSDirectionalEdgeInsets(
+            top: buttonVerticalInset,
+            leading: buttonHorizontalInset,
+            bottom: buttonVerticalInset,
+            trailing: buttonHorizontalInset
+        )
     }
 
+    private var highlightedBackgroundColor: UIColor!
+    private var normalBackgroundColor: UIColor!
+    private var foregroundColor: UIColor!
+
     override init(frame: CGRect) {
         super.init(frame: frame)
 
-        titleLabel?.font = DefaultDynamicFontHelper.preferredBoldFont(
-            withTextStyle: .callout,
-            size: UX.buttonFontSize)
+        configuration = UIButton.Configuration.filled()
         layer.cornerRadius = UX.buttonCornerRadius
         titleLabel?.textAlignment = .center
         titleLabel?.adjustsFontForContentSizeCategory = true
-        contentEdgeInsets = UIEdgeInsets(top: UX.buttonVerticalInset,
-                                         left: UX.buttonHorizontalInset,
-                                         bottom: UX.buttonVerticalInset,
-                                         right: UX.buttonHorizontalInset)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func updateConfiguration() {
+        guard var updatedConfiguration = configuration else {
+            return
+        }
+
+        switch state {
+        case [.highlighted]:
+            updatedConfiguration.background.backgroundColor = highlightedBackgroundColor
+        default:
+            updatedConfiguration.background.backgroundColor = normalBackgroundColor
+        }
+        updatedConfiguration.baseForegroundColor = foregroundColor
```

Comment:
Setting the foreground color like this results in the incorrect color in the highlight state. You can fix this by setting the `foregroundColor` using the `titleTextAttributesTransformer`. You will however have to set the font in the same closure as otherwise it won't be correct. You can then remove setting the font in `configure(viewModel:)`.

Reply:
Ah yes indeed. I came up with a way to still set this closure in `configure(viewModel:)` by using `updatedConfiguration.baseForegroundColor` as `foregroundColor` of the container in the closure, then we can just update the configuration's `baseForegroundColor` value in `updateConfiguration()`.

Kind of like we're already doing with the background color and overriding the transformer closure so it doesn't get modified. That way, the font is still configured in `configure` and it's still all kept in one single place. Tell me if you like it :)

Reply:
(in the case of those two buttons it's a bit trivial because foreground color currently never changes but that way it would still work even with a foreground color based on state changes)